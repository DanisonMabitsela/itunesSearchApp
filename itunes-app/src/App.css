.container {
  max-width: 800px;
  margin: 0 auto;
  padding: 20px;
}

h1 {
  text-align: center;
  margin-bottom: 40px;
}

.search-container {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 40px;
}

.search-container input {
  width: 300px;
  height: 40px;
  border: none;
  border-radius: 20px 0 0 20px;
  padding: 0 20px;
  font-size: 18px;
  box-shadow: 0px 0px 5px #ccc;
  outline: none;
}

.search-container select {
  height: 40px;
  border: none;
  border-radius: 0 20px 20px 0;
  padding: 0 20px;
  font-size: 18px;
  box-shadow: 0px 0px 5px #ccc;
  outline: none;
}

.search-container button {
  height: 40px;
  padding: 0 20px;
  border: none;
  border-radius: 20px;
  font-size: 18px;
  font-weight: bold;
  color: #fff;
  background-color: #0074d9;
  box-shadow: 0px 0px 5px #ccc;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.search-container button:hover {
  background-color: #004c8c;
}

.search-results {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
}

.search-result {
  border: 1px solid #ccc;
  border-radius: 5px;
  overflow: hidden;
}

.search-result img {
  width: 100%;
  height: 200px;
  object-fit: cover;
}

.search-result-info {
  padding: 20px;
}

.search-result-info h3 {
  margin-top: 0;
  margin-bottom: 10px;
  font-size: 24px;
}

.search-result-info p {
  margin-top: 0;
  margin-bottom: 20px;
  font-size: 18px;
}

.search-result-info button {
  height: 40px;
  padding: 0 20px;
  border: none;
  border-radius: 20px;
  font-size: 18px;
  font-weight: bold;
  color: #fff;
  background-color: #0074d9;
  box-shadow: 0px 0px 5px #ccc;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.search-result-info button:hover {
  background-color: #004c8c;
}

.favourites-container {
  margin-top: 40px;
}

.favourites-container h2 {
  font-size: 24px;
  margin-bottom: 20px;
}

.favourite {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  border: 1px solid #ccc;
  border-radius: 5px;
  overflow: hidden;
}
.favourite-info {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
}
.favourite img {
  width: 100px;
  height: 100px;
  object-fit: cover;
}

.favourite-info {
  flex: 1;
  padding: 20px;
}

.favourite-info h3 {
  margin-top: 0;
  margin-bottom: 20px;
  font-size: 18px;
}
.favourite-info button {
  height: 40px;
  padding: 0 20px;
  border: none;
  border-radius: 20px;
  font-size: 18px;
  font-weight: bold;
  color: #fff;
  background-color: #0074d9;
  box-shadow: 0px 0px 5px #ccc;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}
.favourite-info button:hover {
  background-color: #004c8c;
}
