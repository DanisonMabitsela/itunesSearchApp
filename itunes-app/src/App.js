import React, { useState } from "react";
import "./App.css";

function App() {
  // Define state variables using the useState hook
  const [searchTerm, setSearchTerm] = useState("");
  const [searchMedia, setSearchMedia] = useState("all");
  const [searchResults, setSearchResults] = useState([]);
  const [favourites, setFavourites] = useState([]);

  // Define the handleSearch function, which is called when the search button is clicked
  const handleSearch = async () => {
    try {
      // Make a request to the back-end API using fetch
      const response = await fetch(
        `http://localhost:5000/search/${searchTerm}/${searchMedia}`
      );
      // Convert the response to JSON
      const data = await response.json();
      // Update the search results state variable with the results from the API
      setSearchResults(data.results);
    } catch (error) {
      // Log any errors to the console
      console.error(error);
    }
  };

  // Define the handleAddToFavourites function, which is called when the "Add to Favourites" button is clicked
  const handleAddToFavourites = (result) => {
    // Add the clicked result to the favourites array using the spread operator
    setFavourites([...favourites, result]);
  };

  // Define the handleRemoveFromFavourites function, which is called when the "Remove from Favourites" button is clicked
  const handleRemoveFromFavourites = (result) => {
    // Remove the clicked result from the favourites array using the filter method
    setFavourites(
      favourites.filter((favourite) => favourite.trackId !== result.trackId)
    );
  };

  return (
    <div className="container">
      <h1>iTunes Search</h1>
      <div className="search-container">
        <input
          type="text"
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
        <select
          value={searchMedia}
          onChange={(e) => setSearchMedia(e.target.value)}
        >
          <option value="all">All</option>
          <option value="movie">Movies</option>
          <option value="podcast">Podcasts</option>
          <option value="music">Music</option>
          <option value="audiobook">Audiobooks</option>
          <option value="shortFilm">Short Films</option>
          <option value="tvShow">TV Shows</option>
          <option value="software">Software</option>
          <option value="ebook">eBooks</option>
        </select>
        <button onClick={handleSearch}>Search</button>
      </div>
      {/* Display the search results and provide a button to add each result to the favourites list */}
      <div className="search-results">
        {searchResults.map((result) => (
          <div key={result.trackId} className="search-result">
            <img src={result.artworkUrl100} alt={result.trackName} />
            <div className="search-result-info">
              <h3>{result.trackName}</h3>
              <p>{result.artistName}</p>
              <button onClick={() => handleAddToFavourites(result)}>
                Add to Favourites
              </button>
            </div>
          </div>
        ))}
      </div>
      {/* Display the favourites and provide a button to remove each result to the favourites list */}
      <div className="favourites-container">
        <h2>Favourites</h2>
        {favourites.map((favourite) => (
          <div key={favourite.trackId} className="favourite">
            <img src={favourite.artworkUrl100} alt={favourite.trackName} />
            <div className="favourite-info">
              <h3>{favourite.trackName}</h3>
              <p>{favourite.artistName}</p>
              <button onClick={() => handleRemoveFromFavourites(favourite)}>
                Remove from Favourites
              </button>
            </div>
          </div>
        ))}
      </div>
      <a
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      ></a>
    </div>
  );
}

export default App;
