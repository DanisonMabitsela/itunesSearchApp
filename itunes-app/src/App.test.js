import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from "react-test-renderer";
import { getSongs } from "./api";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

describe("App component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("getSongs function", () => {
  it("returns the expected results", async () => {
    const songs = await getSongs();
    expect(songs).toEqual([
      {
        name: "The Beatles",
        artist: "John Lennon",
        album: "Abbey Road",
        year: 1969,
      },
      {
        name: "Michael Jackson",
        artist: "Michael Jackson",
        album: "Thriller",
        year: 1982,
      },
      {
        name: "Queen",
        artist: "Queen",
        album: "Bohemian Rhapsody",
        year: 1975,
      },
    ]);
  });
});
