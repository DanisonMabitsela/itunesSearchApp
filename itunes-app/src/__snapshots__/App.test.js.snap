// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="container"
>
  <h1>
    iTunes Search
  </h1>
  <div
    className="search-container"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <select
      onChange={[Function]}
      value="all"
    >
      <option
        value="all"
      >
        All
      </option>
      <option
        value="movie"
      >
        Movies
      </option>
      <option
        value="podcast"
      >
        Podcasts
      </option>
      <option
        value="music"
      >
        Music
      </option>
      <option
        value="audiobook"
      >
        Audiobooks
      </option>
      <option
        value="shortFilm"
      >
        Short Films
      </option>
      <option
        value="tvShow"
      >
        TV Shows
      </option>
      <option
        value="software"
      >
        Software
      </option>
      <option
        value="ebook"
      >
        eBooks
      </option>
    </select>
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="search-results"
  />
  <div
    className="favourites-container"
  >
    <h2>
      Favourites
    </h2>
  </div>
  <a
    href="https://reactjs.org"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn React
  </a>
</div>
`;
